src/mintpy/los2slope.py

```diff
@@ -0,0 +1,373 @@
+    driver = gdal.GetDriverByName("MEM")
+    cols, rows = dem_array.shape
+    geotransform = (
+        float(meta["X_FIRST"]),
+        float(meta["X_STEP"]),
+        0,
+        float(meta["Y_FIRST"]),
+        0,
+        float(meta["Y_STEP"]),
+    )
+    dataset = driver.Create("dem.tif", rows, cols, 1, gdal.GDT_Float32)
+    dataset.SetGeoTransform(geotransform)
+    dataset.SetProjection("EPSG:4326")
+    band = dataset.GetRasterBand(1)
+    band.WriteArray(dem_array)
+
+    # Calculate slope, save it as slope.tif as DEMProcessing does not allow mem rasters
+    slope_ds = gdal.DEMProcessing("slope.tif", dataset, "slope", scale=111120)
+    slope_array = slope_ds.ReadAsArray()
+
+    # Calculate aspect save it as aspect.tif as DEMProcessing does not allow mem rasters
+    aspect_ds = gdal.DEMProcessing(
+        "aspect.tif",
+        dataset,
+        "aspect",
+        zeroForFlat=True,
+    )
+    aspect_array = aspect_ds.ReadAsArray()
+
+    # Close all rasters
+    slope_ds = None
+    aspect_ds = None
+    dataset = None
+    return slope_array, aspect_array
+
+
+def get_design_matrix4slope(los_inc_angle, los_az_angle, slope, aspect):
+    """Design matrix G to convert asc/desc range displacement into slope direction.
+    Only 2D approach is implemented.
+
+    Project displacement from LOS to slope direction components:
+
+    Math for 2D:
+        dLOS =   dslope * G
+        where, G = (sin(inc_angle) * cos(az_angle - aspect) * cos(slope) + cos(inc_angle) * sin(slope))
+
+    Parameters: los_inc_angle - 1/2D np.ndarray in size of (num_file, window_y, window_x), LOS incidence angle in degree.
+                los_az_angle  - 1/2D np.ndarray in size of (num_file, window_y, window_x), LOS azimuth angle in degree, measured in anti-clockwise direction from east.
+                slope         - 1/2D np.ndarray, slope of the terrain in degree.
+                aspect        - 1/2D np.ndarray, aspect of the terrain in degree, measured in anti-clockwise direction from east.
+    Returns:    G             - 2D matrix
+    """
+    # 2D approach: not currently working, needs math re-check
+    # aspect = -aspect
+    # vh = np.sin(np.deg2rad(los_inc_angle))*np.cos(np.deg2rad(los_az_angle-aspect))
+    # vu = np.cos(np.deg2rad(los_inc_angle))
+    # ch = np.cos(np.deg2rad(slope))
+    # cu = np.sin(np.deg2rad(slope))
+    # G = vh*ch + vu*cu
+
+    # # 3d Approach
+    aspect = (
+        -aspect + 90
+    )  # Shift hillsope aspect to 0 east anti-clockwise, same as the azimuth angle.
+
+    ve = np.sin(np.deg2rad(los_inc_angle)) * np.sin(np.deg2rad(los_az_angle)) * -1
+    vn = np.sin(np.deg2rad(los_inc_angle)) * np.cos(np.deg2rad(los_az_angle))
+    vu = np.cos(np.deg2rad(los_inc_angle))
+
+    ce = (np.sin(np.deg2rad(aspect)) * np.sin(np.deg2rad(slope))) * -1
+    cn = np.sin(np.deg2rad(aspect)) * np.cos(np.deg2rad(slope))
+    cu = np.cos(np.deg2rad(slope))
+
+    G = ve * ce + vu * cu + vn * cn
```

Comment:
Here, 3 components of the unit_vector are sum up, I think projection to slope should be done differently:
on the example for East_West:
                        hv_proj.    *   vt_proj 
ve = -sin(azimuth_angle) * sin(inc_angle), 

you would use above to project LOS to EW and assume vn == vu =0. Hv_proj projects the vector onto EW direction, and vt_proj projects in on the horizontal plane. So I think you should project hv_proj to the slope aspect direction, and vt_proj down slope: e.g  hv_proj = cos(azimuth + aspect) * -1 , vt_proj= cos(slope + incidence), and downslope = dlos * hv_proj * vt_proj. Please double check the math

Reply:
@mgovorcin The assumption that Ashok is using is different than projecting LOS first to horizontal and then to the downslope direction. The assumption here is that all of the displacement is parallel to the downslope direction and we are only seeing a part of that in the LOS direction, so the result is larger than the LOS value.

Reply:
I think this could be tested using a simplified example that can be added to the tests directory as @yunjunz suggested. I made a rough example of this using [asc_desc2horz_vert.py](https://github.com/insarlab/MintPy/blob/main/tests/asc_desc2horz_vert.py) as reference. (example attached)
[los2slope_test.py.txt](https://github.com/user-attachments/files/16060911/los2slope_test.py.txt)